request-server.go

```diff
@@ -33,20 +33,50 @@ type RequestServer struct {
 }
 
 // NewRequestServer creates/allocates/returns new RequestServer.
-// Normally there there will be one server per user-session.
+// Normally there will be one server per user-session.
 func NewRequestServer(rwc io.ReadWriteCloser, h Handlers) *RequestServer {
+	rs, _ := NewRequestServerWithOptions(rwc, h, nil...)
```

Comment:
You don’t need to specify: `nil...`, you just don’t have to include any options at all.

Is there a reason to split up `NewRequestServer` and `NewRequestServerWithOptions`? Unless we expect that people are assigning this into a variable with a specific type, then there should not be no reason why we couldn’t use `NewRequestServer(io.ReadWriteCloser, Handlers, ...RequestServerOption) (*RequestServer, error)`

Reply:
> You don’t need to specify: `nil...`, you just don’t have to include any options at all.
> 

you are right, sorry for that

> Is there a reason to split up `NewRequestServer` and `NewRequestServerWithOptions`? Unless we expect that people are assigning this into a variable with a specific type, then there should not be no reason why we couldn’t use `NewRequestServer(io.ReadWriteCloser, Handlers, ...RequestServerOption) (*RequestServer, error)`

the reason is backward compatibility, currently `NewRequestServer` does not return an error while `NewServer` returns `(*Server, error)`, the error can be returned there only when setting an option and it can never happen with the current code.
So for `NewRequestServer`:

1) we can break compatibility using the method you suggest
2) introduce a new method as I did
3) don't allow the RequestServerOption to return an error: this should work with current code but could not work in future

Reply:
Right, but the only actual option available right now, cannot return a non-`nil` error. So, do we really need to build having an `error` return value into the `RequestServerOption`?

Reply:
Ok, done, thanks!